improver/blending/calculate_weights_and_blend.py

```diff
@@ -240,13 +243,39 @@ def process(
             if spatial_weights:
                 weights = self._update_spatial_weights(cube, weights, fuzzy_length)
 
+            dims_to_collapse = set(range(cube.ndim)) - {
+                cube.coord_dims(self.blend_coord)[0]
+            }
+
+            # In the event of multiple dimensions, always use the first available one to slice over
+            dim_to_collapse = min(dims_to_collapse)
+            number_of_subcubes = cube.shape[dim_to_collapse]
+            indices = [slice(None)] * cube.ndim
+
             # blend across specified dimension
             BlendingPlugin = WeightedBlendAcrossWholeDimension(self.blend_coord)
-            result = BlendingPlugin(
-                cube,
-                weights=weights,
-                cycletime=cycletime,
-                attributes_dict=attributes_dict,
+
+            cubes = []
+            for subcube_index in range(number_of_subcubes):
+                indices[dim_to_collapse] = subcube_index
+                subcube = cube[tuple(indices)]
+                result = BlendingPlugin(
+                    subcube,
+                    weights=weights,
+                    cycletime=cycletime,
+                    attributes_dict=attributes_dict,
+                )
+                result.data = result.lazy_data()
+                cubes.append(result)
+            result = CubeList(cubes).merge_cube()
+
+            # Making the dim coords consistent between the original cube and new blended cube
+            dim_coords = {coord.name() for coord in cube.dim_coords}
+            blend_dim = {cube.coords()[cube.coord_dims(self.blend_coord)[0]].name()}
+            promote_set = (
+                dim_coords - {coord.name() for coord in result.dim_coords} - blend_dim
             )
+            for dim in promote_set:
+                result = new_axis(result, dim)
```

Comment:
Unfortunately, this logic breaks for percentile blending (it will try to slice over the percentile coord, which is not allowed). I think this could also be simplified somewhat by knowing that blending dim is always 0:
```python
            # Note: to reduce memory, slice over the first non-blended dim
            # and merge afterwards (blended dim is 0)
            slice_dim = 1
            allow_slicing = (
                cube.ndim > 1
                and cube.shape[slice_dim] > 1
                and not BlendingPlugin.check_percentile_coord(cube)
            )
            cube_slices = cube.slices_over(slice_dim) if allow_slicing else [cube]
            result_slices = CubeList()
            for cube_slice in cube_slices:
                result_slice = BlendingPlugin(
                    cube_slice,
                    weights=weights,
                    cycletime=cycletime,
                    attributes_dict=attributes_dict,
                )
                result_slices.append(result_slice)
            result = result_slices.merge_cube() if allow_slicing else result_slices[0]
```

Note that the above code explicitly excludes percentile cubes from slicing. In the future, it would be nice to to fold percentile and non-percentile blending into a single code path inside `WeightedBlendAcrossWholeDimension` (the difference is mostly in the aggregator used and the code structure is otherwise pretty much the same, CC: @fionaRust).